### DISCUSSION TITLE
template appending to pinned note instead of creating new note

### DISCUSSION THREAD

**Discussion by danieltomasz** [SilentVoid13/Templater#561] [Help]
Do I have something wrong or this is a bug?
```
<%*
let qcFileName= await tp. system.prompt("Note Title")
titleName = qcFileName 
await tp.file.rename(titleName)
await tp.file.move ("/Notes/"+ titleName);
-%>
---
title: "<% qcFileName %>"
author: Daniel Borek
date: <% tp.date. now("YYYY-MM-DD")%>
tags: [note]
---
<% tp.file.cursor() %>
```

**Comment by AB1908** [comment]
We'll need some more detailed steps than that. Hard to go off of this.

**Reply by philoserf** [reply]
Right off the bat, I see a space here `tp. system` in line 2

My rules:
- if a thing has a name do not give it another name
- if it is not required do not include it

```
<%*
let title = await tp.system.prompt("Title");
await tp.file.move("/Notes/"+ title);
%>
---
title: <% title %>
author: Daniel Borek
date: <% tp.date.now("YYYY-MM-DD") %>
tags: [note]
---
<% tp.file.cursor() %>
```

Note: I have almost no Templater experience, yet.
Note: I think this does what you wanted, thanks for the learning challenge.
Tested: @danieltomasz 

Let me know if any of my changes to your example require a better explanation, please.

**Comment by danieltomasz** [comment]
@philoserf indeed there was an unnecessary space (I did OCRed this code from someone screenshot),  thanks!
 this new code works better and doesn't append to the current pinned note but this time unpin it and close.

Steps to reproduce
1. Add to vault this template as '/Templates/Note'
```
<%*
let title = await tp.system.prompt("Title");
await tp.file.move("/Notes/"+ title);
%>
---
title: <% title %>
author: Daniel Borek
date: <% tp.date.now("YYYY-MM-DD") %>
tags: [note]
---
<% tp.file.cursor() %>
```
2. In the command palette I  pick "Create New Note from Template and I pick note
3. The newly created note replace currently open pinned  note in  active pane

this is expected behaviour when the active note is not pinned,  but I would expect that the new note will be open in the second/newly split  pane and the pinned note stays open

Templater Version: 1.10.0
My debug info
```
SYSTEM INFO:
	Obsidian version: v0.13.23
	Installer version: v0.12.19
	Operating system: Darwin Kernel Version 21.4.0: Mon Feb  7 21:32:19 PST 2022; root:xnu-8020.100.429.131.1~2/RELEASE_ARM64_T8101 21.4.0
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: Minimal
	Snippets enabled: 0
	Safe mode: off
	Plugins installed: 3
	Plugins enabled: 3
		1: Minimal Theme Settings
		2: Style Settings
		3: Templater

RECOMMENDATIONS:
	Custom theme: for cosmetic issues, please first try updating your theme to latest. If still not fixed, please try to make the issue happen in the help vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the help vault or disable community plugins.
``

**Reply by AB1908** [reply]
Wait, create new note from template isn't Templater functionality, is it?

**Reply by danieltomasz** [reply]
![image](https://user-images.githubusercontent.com/7980381/154558232-93da644d-7a7f-4ad3-973f-d8d59c5ae0be.png)

**Reply by AB1908** [reply]
Anyhow, I'll try and repro tomorrow and report back.

**Reply by AB1908** [reply]
Haha I see. I guess I don't use Templater as much as I thought I did.

**Reply by AB1908** [reply]
I'm so sorry. I swear I'm working on better task management. I'll try to repro this over the weekend and report back.

**Reply by AB1908** [reply]
Hmm, I did think of a few ways around this but haven't tested. What is the overall use case? I feel like we're going about this the wrong way.

**Reply by danieltomasz** [reply]
I  created a button in the sidebar for creating specific types of new notes;
it would be helpful for my workflow if I would be able to pin some notes (like references) and to create new notes that will open in the side-pane
But indeed, maybe I am doing it the wrong way, and I could achieve my type of workflow with "quick add"  instead of templater for example (which I don't know to set up to achieve needed results yet)

**Reply by AB1908** [reply]
Hmm creating new notes in the side pane sounds tricky. Would need to interface with some CodeMirror APIs I think. You can use QuickAdd to append to or create new notes in the background. It also has an option to open the note iirc.

**Reply by danieltomasz** [reply]
> Hmm creating new notes in the side pane sounds tricky. Would need to interface with some CodeMirror APIs I think. You can use QuickAdd to append to or create new notes in the background. It also has an option to open the note iirc.

Yes, maybe this is the way
